iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java

```diff
@@ -0,0 +1,62 @@
+package eu.dnetlib.iis.core.spark.pipe;
+
+import java.io.Serializable;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.spark.SparkFiles;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+
+import eu.dnetlib.iis.core.common.AvroGsonFactory;
+import scala.Tuple2;
+
+
+/**
+ * Executor of mapreduce scripts using spark pipes.
+ * It imitates hadoop streaming behavior. 
+ * 
+ * @author madryk
+ *
+ */
+public class SparkPipeExecutor implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+
+	//------------------------ LOGIC --------------------------
+	
+	/**
+	 * Imitates map part of hadoop streaming job
```

Comment:
The comment does not mean much for people who do not know hadoop streaming. And even those would not know that this method assumes that the invoked script must return lines containing tab separated key/values and takes and operates on keys.